secboot/secboot_sb_test.go

```diff
@@ -1023,13 +1033,23 @@ func (s *secbootSuite) TestResealKey(c *C) {
 		})
 		defer restore()
 
+		// mock ReadSealedKeyObject
+		readSealedKeyObjectCalls := 0
+		mockSealedKeyObjects := []*sb_tpm2.SealedKeyObject{{}, {}}
```

Comment:
Unlike auth file, the keyfile/keyfile2 are just strings rather than files that can be opened and read into SealedKeyObject struct. When I tried to create more full SealedKeyObjects, it was not allowed, as none of the fields are exported. I guess we need to stop having keyfile/keyfile2 as just strings, and have them as actual files, write them out to temp directory, to read them and process them into SealedKeyObjects?

Or is there some other API i can use to generate and created SealedKeyObject?

Reply:
The error is already on the tc, and its being mocked and tested.

I can improve mockSealedKeyObjects to be as goot ad mockTPMPolicyAuthKey which is not on tc either.